weather.py

```diff
@@ -2,116 +2,106 @@
+                        break
                     else:
-                        if int(line[1]) >= mx_tmp:
-                            mx_tmp = int(line[1])
-                            mx_dt = line[0]
-                        if int(line[3]) <= mn_tmp:
-                            mn_tmp = int(line[3]) 
-                            mn_dt = line[0]
-                        if int(line[7]) >= mx_hum:
-                            mx_hum = int(line[7]) 
-                            hum_dt = line[0]
+                        if line[1] == '' and line[3] == '' and line[7] == '':
+                            continue
+                        else:
+                            if int(line[1]) >= max_tmp:
+                                max_tmp = int(line[1])
+                                maxtmp_dt = line[0]
+                            if int(line[3]) <= min_tmp:
+                                min_tmp = int(line[3]) 
+                                mintmp_dt = line[0]
+                            if int(line[7]) >= max_hum:
+                                max_hum = int(line[7]) 
+                                maxhum_dt = line[0]
 
+    c_maxtmp_dt = string_to_date(maxtmp_dt)
+    c_mintmp_dt = string_to_date(mintmp_dt)
+    c_maxhum_dt = string_to_date(maxhum_dt)
 
-    def convert_list_to_date(dt):
-        ''.join(dt)
-        dt_obj = datetime.strptime(dt,'%Y-%m-%d')
-        c_dt = dt_obj.strftime("%b %d")
-        return c_dt
+    return [max_tmp,c_maxtmp_dt,min_tmp,c_mintmp_dt,max_hum,c_maxhum_dt]
 
-    # converted_dt = convert_list_to_string(mx_dt)
-    # dt_obj = datetime.strptime(converted_dt,'%Y-%m-%d')
-    # c_dt = dt_obj.strftime("%b %d")
-    cmx_dt = convert_list_to_date(mx_dt)
-    cmn_dt = convert_list_to_date(mn_dt)
-    chum_dt = convert_list_to_date(hum_dt)
+def string_to_date(dt):
+    dt_obj = datetime.strptime(dt,'%Y-%m-%d')
+    c_dt = dt_obj.strftime("%b %d")
+    return c_dt
 
-    # print(f"Highest: {mx_tmp}C on {cmx_dt}")
-    # print(f"Lowest: {mn_tmp}C on {cmn_dt}")
-    # print(f"Humidity: {mx_hum}% on {chum_dt}")
-    return [mx_tmp,cmx_dt,mn_tmp,cmn_dt,mx_hum,chum_dt]
 
-def monthcalc(m):
-    highest_tmp = 0
-    lowest_tmp = 0
-    mean_hum = 0
-    cnt = 0
-    str_mn = ''.join(m)
-    dt = datetime.strptime(str_mn, "%Y%m")
+def month_calculator(mo):
+    sum_high_tmp = 0
+    sum_low_tmp = 0
+    sum_mean_hum = 0
+    count = 0
+    str_mo = ''.join(mo)
+    dt = datetime.strptime(str_mo, "%Y%m")
     c_dt = str(dt.strftime("%Y_%b"))
-    os.chdir(r'C:\Som\Personal Projects\weatherdata')
-    for f in os.listdir():
-        #print(f)
-        if f.__contains__(c_dt):
-            with open (f'C:\\Som\\Personal Projects\\weatherdata\\{f}','r') as data_file:
-
+    os.chdir(data_folder)
+    for file in os.listdir():
+        if c_dt in file:
+            with open (f'{data_folder}/{file}','r') as data_file:
                 csv_data = csv.reader(data_file, delimiter=',')
```

Comment:
Why you did not use csv Dictreader?

Reply:
Yes, i read about Dictreader. will implement that and see

Reply:
I tried with Dictreader but got some key error hence i had used reader only

Reply:
Can you try dictreader now?